hoomd/operation.py

```diff
@@ -94,7 +94,10 @@ def __setattr__(self, attr, value):
         elif attr in self._typeparam_dict.keys():
             self._setattr_typeparam(attr, value)
         else:
-            super().__setattr__(attr, value)
+            self._setattr_hook(attr, value)
```

Comment:
So this is complicated (potentially too much so), but the idea is that the first `if` is to ensure that if `_param_dict`, `_typeparam_dict`, or `_cpp_obj` are not set we set them (we have to use a class level variable to define which attributes to do this for) otherwise when attempting to set those attributes you get things like `obj` has no attribute `_param_dict`  when attempting to set in a constructor. I know you can just call `super().__setattr__` instead, but there is no `__init__` method for `_Operation` (`super().__init__` doesn't work because derived classes sometimes need their `__init__` called at the end of their child classes' constructors).

We can remove the `_use_default_setattr` and just use that in the `_setattr_hook`.

Reply:
Maybe I'm confused. It seems like the logic you're referring to is the `setattr` call in the `__getattr__` method, which ensures that the special keys get set to their default values the first time they're accessed. However, in `__setattr__` the first `if` clause's logic is _identical_ to the logic in `_setattr_hook`. Are you suggesting that you need to be able to override only this behavior in subclasses, but not the behavior of the first `if` statement?

Reply:
Yes, that is what I am suggesting; however, I found a solution, I think to be better. In the `__getattr__` method just call `object.__setattr__(attr, value)` in those cases. Then it works without the first clause for that case. We still need it for now with `triggers` since they both exist in the `_param_dict` and need special treatment for now (C++ `Updater` and `Analyzer` do not have triggers and are stored in a `std::pair` in the C++ `System`, this will change eventually and the need for that code will be complete removed.